### DISCUSSION TITLE
Inference mode of 3D Generator

### DISCUSSION THREAD

**Discussion by Ismailok** [Project-MONAI/MONAI#5060] [Q&A] [answered]
Hello MONAI team ! 

I am working on 3DGAN to generate 3D volumes from a noise vector. I used the transformers of monai like resize(96,96,96) to train the model but basically the dimensions of my data (512,512,350).
Once the training of the is done, the model generates volumes with size (96,96,96). At the inference mode step, I did not find a solution for a 2D input (noise vector) and the output 3D volumes with size (512,512,350).
Does anyone have a solution or suggestion, please ?

Best regards,

**Comment by ebrahimebrahim** [comment]
"Noise vector" means input to some kind of decoder? Why is the noise vector 2D?
(If it doesn't have to be 2D then you can reshape the noise vector to fit whatever it needs to go into)

**Comment by Ismailok** [comment, marked as answer]
Yes, the "noise vector" is the input to the decoder, it is like 
"noise_vector = torch.randn(1, 1000).to(device)" and the model is a 3D volume decoder (generator) from the noise vector. then, "decoder(noise_vector)" gives a 3D volumes of dimension (96,96,96).

**Reply by ebrahimebrahim** [reply]
Oh so inputting noise vector is fine for you-- then is the problem that you are getting output of dimension (96,96,96) but you want output of dimension (512,512,350)? You can upsample using the monai resize transform, for example setting it to trilinear interpolation to get something with more voxels but very blurry. If you want to truly generate images at that resolution then your model has to be designed and trained for that in the first place. Or there are also super resolution GAN techniques to look into
